src/hyperion/device_setup_plans/read_hardware_for_setup.py

```diff
@@ -19,24 +20,26 @@ def read_hardware_for_ispyb_pre_collection(
     undulator: Undulator,
     synchrotron: Synchrotron,
     s4_slit_gaps: S4SlitGaps,
+    robot: BartRobot,
```

Comment:
Changing the signature of these methods causes so many other code changes due to the number of callers. Perhaps this is trying to tell us something.
Do we actually need all the different types of composite?

Reply:
I agree this is something to think about but seems outside the scope of this issue. I've made https://github.com/DiamondLightSource/hyperion/issues/1128